diff --git a/modulbin.c b/modulbin.c
index 6dea4a5..4f70d44 100755
--- a/modulbin.c
+++ b/modulbin.c
@@ -124,20 +124,23 @@ void UpdateDefmoduleItemHeader(
 void UpdateDefmoduleItemHeaderHM(
   Environment *theEnv,
   struct bsaveDefmoduleItemHeader *theBsaveHeader,
   struct defmoduleItemHeaderHM *theHeader,
   size_t itemSize,
   void *itemArray)
   {
    size_t firstOffset, lastOffset;

    theHeader->theModule = ModulePointer(theBsaveHeader->theModule);
+   theHeader->itemCount = 0;
+   theHeader->hashTableSize = 0;
+   theHeader->hashTable = NULL;
    if (theBsaveHeader->firstItem == ULONG_MAX)
      {
       theHeader->firstItem = NULL;
       theHeader->lastItem = NULL;
      }
    else
      {
       firstOffset = itemSize * theBsaveHeader->firstItem;
       lastOffset = itemSize * theBsaveHeader->lastItem;
       theHeader->firstItem =
